operators/src/engine/operator.rs

```diff
@@ -149,6 +149,17 @@ pub trait InitializedRasterOperator: Send + Sync {
 
     /// Get a canonic representation of the operator and its sources
     fn canonic_name(&self) -> CanonicOperatorName;
+
+    /// Get the unique name of the operator
+    fn name(&self) -> &'static str;
+
+    // Get the path of the operator in the workflow
+    fn path(&self) -> WorkflowOperatorPath;
+
+    /// Return the name of the data loaded by the operator (if any)
+    fn data(&self) -> Option<String> {
```

Comment:
`Option<&String>` or `Option<Arc<str>>` for cheap copy?

Reply:
doing `Option<&String>` won't help much because we need to send the result down a channel, which requires owning it.

`Option<Arc<str>>`  would work I guess. But is the synchronization overhead really better than the cloning? The String will always be small.